File: Examples/VisualBasic/Attachments/AttachFileAndSetIcon.vb
Imports System.IO
Imports AsposeNote = Aspose.Note
Imports System.Collections.Generic
Imports System.Drawing.Imaging
Namespace Attachments
    Public Class AttachFileAndSetIcon
        Public Shared Sub Run()
            ' ExStart:AttachFileAndSetIcon
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Attachments()

            ' Create an object of the Document class
            Dim doc As New Document()
            ' Initialize Page class object
            Dim page As New AsposeNote.Page(doc)
            ' Initialize Outline class object
            Dim outline As New Outline(doc)
            ' Initialize OutlineElement class object
            Dim outlineElem As New OutlineElement(doc)
            ' Initialize AttachedFile class object and also pass its icon path
            Dim attachedFile As New AttachedFile(doc, dataDir & Convert.ToString("attachment.txt"), File.OpenRead(dataDir & Convert.ToString("icon.jpg")), ImageFormat.Jpeg)
            ' Add attached file
            outlineElem.AppendChild(attachedFile)
            ' Add outline element node
            outline.AppendChild(outlineElem)
            ' Add outline node
            page.AppendChild(outline)
            ' Add page node
            doc.AppendChild(page)

            dataDir = dataDir & Convert.ToString("AttachFileAndSetIcon_out_.one")
            doc.Save(dataDir)
            ' ExEnd:AttachFileAndSetIcon      

            Console.WriteLine(Convert.ToString(vbLf & "File attached and it's icon setup successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace

File: Examples/CSharp/Attachments/AttachFileAndSetIcon.cs
using System.IO;
using System;
using System.Drawing.Imaging;

namespace Aspose.Note.Examples.CSharp.Attachments
{
    public class AttachFileAndSetIcon
    {
        public static void Run()
        {
            // ExStart:AttachFileAndSetIcon
            // ExFor:Document
            // ExFor:AttachedFile
            // ExSummary:Shows how to add a file from a stream to a document.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Attachments();

            // Create an object of the Document class
            Document doc = new Document();
            
            // Initialize Page class object
            Page page = new Page();
            
            // Initialize Outline class object
            Outline outline = new Outline();
            
            // Initialize OutlineElement class object
            OutlineElement outlineElem = new OutlineElement();

            using (var stream = File.OpenRead(dataDir + "icon.jpg"))
            {
                // Initialize AttachedFile class object and also pass its icon path
                AttachedFile attachedFile = new AttachedFile(dataDir + "attachment.txt", stream, ImageFormat.Jpeg);

                // Add attached file
                outlineElem.AppendChildLast(attachedFile);
            }

            // Add outline element node
            outline.AppendChildLast(outlineElem);
            
            // Add outline node
            page.AppendChildLast(outline);
            
            // Add page node
            doc.AppendChildLast(page);

            dataDir = dataDir + "AttachFileAndSetIcon_out.one";
            doc.Save(dataDir);

            // ExEnd:AttachFileAndSetIcon      

            Console.WriteLine("\nFile attached and it's icon setup successfully.\nFile saved at " + dataDir);
        }        
    }
}